### ISSUE TITLE
Politeia redesign missing items

### ISSUE THREAD

**Issue by fernandoabolafio** [decred/dcrdesign#159]
This issue is to better track the missing items for the Politeia redesign as the [master issue](https://github.com/decred/dcrdesign/issues/77) has grown long enough to be hard to follow what is going on.

The design items we noticed as missing, so far, are:

- [x] User Account
     - [x] Manage Identity
     - [x] Preferences tab
- [x] Proposal
    - [x] Proposal action buttons (approve, censor, authorize vote to start, revoke vote authorization, start a vote, abandon)
     - [x] Where to click to edit a proposal
     - [x] Proposal version picker 
- [x] Proposal form
   - [x] Attached images display (Perhaps use an image wall and allow users to preview the images using a modal?)
   - [x] Attached images validation errors
   - [x] Attached images policy info (max size, max images, etc)
- [ ] Loading of content: how the loading should be displayed in the different contexts of the website.
- [x] Admin pages: list proposals and search users (and also where to click to reach the admin page)
- [x] Errors: 
     - [x] Global errors (ex failed fetching proposals list)
     - [x] Non-main context errors (ex failed fetching treasure balance). This is the kind of error that 
       needs to be signalized but should not affect the overall UX.

**Update 3/07:**
 - [x] Placeholder for empty proposals lists 

**Update 23/07:**
- [x] Login modal

**Update 06/07:**
- [x] Comments Censor 

**Update 26/08:**
- [x] Highlight new comments since last visit 
- [ ] Indicate pi is running on testnet with a different header or something like that. Original issue: https://github.com/decred/politeiagui/issues/1072/
- [ ] A "splash screen" to be displayed while the politeia is executing it's initial loading. Something like the logo centralized in the middle of the page.

**Comment by tiagoalvesdulce** [comment]
In the User Account, would be nice if we have an example of a success message to show that a rescan was successful, the number of credits added, etc

**Comment by MariaPleshkova** [comment]
What is Proposal version picker?

**Comment by fernandoabolafio** [comment]
@MariaPleshkova, when a proposal has multiple versions, in the proposal detail page, is possible to visualize the old version as well the text diff between two subsequent versions.
Like this:
<img width="918" alt="Screen Shot 2019-07-02 at 12 33 21" src="https://user-images.githubusercontent.com/14864439/60506232-baeb8b80-9cc5-11e9-87b3-822dbd3a2228.png">

<img width="965" alt="Screen Shot 2019-07-02 at 12 33 27" src="https://user-images.githubusercontent.com/14864439/60506219-b0c98d00-9cc5-11e9-9832-b7c8600e682a.png">

**Comment by MariaPleshkova** [comment]
@fernandoabolafio got it, thanks

**Comment by linnutee** [comment]
Commenting here to hold off the work on account and credit purchase flows implementation. 

I've realised there's some simple but significant UX improvements needed there. Essentially around grouping and bringing them into subpages rather than relying on modals this much. Provided Maria with drafts which will be addressed as visual design. 

Also few questions/requests: 
- Forget password: can the process of this be explained a bit. There's presently the reset password token which can be manually expired by the user. Will another one be generated right when this is expired and is there even a good reason expire this? Could this instead happen the same way when logged in or not logged in (eg. github example) that the reset pw token/link is only sent to the email and not displayed in the UI. 

- Regarding Past Public Keys, can this list include the date when they were generated as well a small description of why these are stored + what can they be used for? 

- User ID: also needs a small description/input how exactly this differs from the Public Key.

**Comment by fernandoabolafio** [comment]
@linnutee 
> Forget password: can the process of this be explained a bit. There's presently the reset password token which can be manually expired by the user. Will another one be generated right when this is expired and is there even a good reason expire this? Could this instead happen the same way when logged in or not logged in (eg. GitHub example) that the reset pw token/link is only sent to the email and not displayed in the UI.

The password "reset password token" can only be expired by the admin, not the user. This is also only displayed to the admin. This is an admin tool to handle some edge cases where a user can be stuck with an invalid token. Once a token is expired by the admin, the user will be able to request a new one.
So to make more clear to you what actions are available for the admin and for the user:
- User: change password, change username, manage identity.
- Admin: expire tokens (for the verification email and password reset), deactivate the account and rescan proposal credits.

> Regarding Past Public Keys, can this list include the date when they were generated as well a small description of why these are stored + what can they be used for?

IIRC this list of past identities is being shown only to the logged in user even though the information is public in the API. In any case, I think a help description would be enough here.

> User ID: also needs a small description/input how exactly this differs from the Public Key.

Agreed. Perhaps we can have an "i" icon in front of the id.

**Comment by tiagoalvesdulce** [comment]
@linnutee @MariaPleshkova any update on the proposal credits purchase flow and the account UX improvements?

**Comment by MariaPleshkova** [comment]
@tiagoalvesdulce working on them. Will provide you with the designs soon

**Comment by MariaPleshkova** [comment]
@fernandoabolafio 
Attaching designs with the latest updates.

-**User account modifications** 
Realized the way grouping is presently done has issues, as there are a lot of not related items mashed together. So we decided to make 4 high level groups as subpages and populating those with the relevant items (Identity, Account, Credits, Preferences). + added missing relevant pages.

-**Manage identity modifications**
Reduced one action, so now both buttons (Create new identity and Import identity) could be accessed here already, and there will be no need to double click (first opening modal “Manage identity”, then the action).

-**Added pages from My proposals’ tab** 

-**Added Attach new file to the proposal flow**

-**Back to cyan color for an active tab underlining (#2ED8A)**

-**Added Manage Identity modals**

-**Added proposal actions (censor, approve,etc.) (see the artboard with all proposal cards’ examples)**

-**Improved tabs visualization for smaller screens (2 separate dropdowns for 2 filters)**

-**Added not found page**

-**Added notifications examples**

-**Added an Edit icon to the proposal page (if it’s “my” proposal only, of course)**

P.S. It's still Proposal version picker left.
Regarding admin pages, I thought that admin view is the same page with all proposals, but an admin can see additional buttons on them, like "start vote"/"abandon". And that's it. And to reach the admin view you should just login as an admin. Or am I wrong?

Preview: https://xd.adobe.com/view/a6a62142-18c9-4d0c-4ebd-0fcdec0390ad-d097/
Specs: https://xd.adobe.com/spec/25f9f648-8d13-446f-75aa-242b21116f87-53c8/

**Comment by fernandoabolafio** [comment]
@MariaPleshkova I am taking a look at the new specs, everything looking good so far. The account page is looking much better indeed.

Now, answering your question: 
> Regarding admin pages, I thought that admin view is the same page with all proposals, but an admin can see additional buttons on them, like "start vote"/"abandon". And that's it. And to reach the admin view you should just login as an admin. Or am I wrong?
Those are two different things:
- The admin can see some special buttons in the public list, this is correct.
- The admin has another page which lists non-public proposals (or "unvetted proposals"), see below:
![Screen Shot 2019-07-15 at 08 22 16](https://user-images.githubusercontent.com/14864439/61198316-b8cbf880-a6d9-11e9-8a9b-cf3b194ba98a.png)
Notice that this page has also a "Search Users" button on the right, which takes to a different page as well. Do you have access as an admin on https://test-proposals.decred.org? From there you can go to every path and see how currently works.

**Comment by MariaPleshkova** [comment]
Attaching Pi XD file as you asked @fernandoabolafio 

[Politeia_17.07.2019.zip](https://github.com/decred/dcrdesign/files/3401824/Politeia_17.07.2019.zip)

**Comment by fernandoabolafio** [comment]
@MariaPleshkova just added one more item to the list: Login modal. We need to display that when the user session has expired, so they don't loose context by being redirected to the login (in case they are in a protected route). Or at least this is how we dealt with this situation in the old GUI. Do you have any thoughts on how to improve this UX?

**Comment by MariaPleshkova** [comment]
@fernandoabolafio but we already have this one in the updated xd above:
![image](https://user-images.githubusercontent.com/8841018/61889771-b86c0280-af0e-11e9-8a8f-6ddcb4a02d3b.png)

**Comment by MariaPleshkova** [comment]
@fernandoabolafio @tiagoalvesdulce attaching the links to admin pages + xd file for your convenience
preview: https://xd.adobe.com/view/90a9155f-0071-4e1b-78d1-e56b363473e1-e698/
spec: https://xd.adobe.com/spec/4e7eb732-a445-46b0-4f7e-4152097a10c8-384f/

XD file:  [Politeia_admin_02.08.2019.zip](https://github.com/decred/dcrdesign/files/3461260/Politeia_admin_02.08.2019.zip)

**Comment by fernandoabolafio** [comment]
@MariaPleshkova about the admin specs, I have one question: how the admin can reach the unvetted proposals route? I am supposing it is through the link "Proposals" in the username dropdown. However, the admin is also a regular user so the dropdown should also show "My Proposals". So one option would be show "My Proposals" and "Unvetted Proposals" for example.

**Comment by fernandoabolafio** [comment]
Just added one more thing. Comments censor specs. This is how it looks like currently:

<img width="250" alt="Screen Shot 2019-08-06 at 14 04 02" src="https://user-images.githubusercontent.com/14864439/62538558-65217a80-b853-11e9-942e-97ab50e35080.png">

<img width="775" alt="Screen Shot 2019-08-06 at 14 04 15" src="https://user-images.githubusercontent.com/14864439/62538564-6a7ec500-b853-11e9-9642-fd7bf0de80a7.png">

<img width="204" alt="Screen Shot 2019-08-06 at 14 04 28" src="https://user-images.githubusercontent.com/14864439/62538569-6ce11f00-b853-11e9-985a-246bc70e5348.png">

**Comment by MariaPleshkova** [comment]
@fernandoabolafio 

> So one option would be show "My Proposals" and "Unvetted Proposals" 

Yes. Totally agree!

**Comment by linnutee** [comment]
Hey @fernandoabolafio wondering if there's still any relevant items here amongst the couple open points or can the issue be closed?
